lib/pbio/include/pbdrv/cache.h

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2025 The Pybricks Authors
+
+#ifndef _PBDRV_CACHE_H_
+
+#include <stddef.h>
+
+// Gets data ready so that the data that we (the CPU) have written
+// can be read by DMA peripherals. This cleans the relevant cache lines
+// and flushes the write buffer.
+void pbdrv_cache_prepare_before_dma(const void *buf, size_t sz);
+
+// Makes sure that we (the CPU) are able to read data written
+// by DMA peripherals. This invalidates the relevant cache lines.
+void pbdrv_cache_prepare_after_dma(const void *buf, size_t sz);
+
+// Accesses a variable via the uncached memory alias
+#define PBDRV_UNCACHED(x)           (*(volatile __typeof__(x) *)((uint32_t)(&(x)) + 0x10000000))
```

Comment:
This seems very specific to EV3, so we should rename things accordingly. Or just put this in the tiam1808 library.

Reply:
Definitely not in tiam1808, since it's specific to the way we've specifically programmed the MMU. `PBDRV_EV3_UNCACHED`?